### PR TITLE
feat(ui): add history to WorkflowTemplate & ClusterWorkflowTemplate details

### LINKED ISSUE #13282: UI: Add execution history to WorkflowTemplate & ClusterWorkflowTemplate details
# Summary

It would be useful to have the execution history as well as the details of the CronWorkflow.
It is a bit tedious to have to search through the workflow each time.

## Use Cases

Used to check the execution history of a specific WorkflowTemplate or ClusterWorkflowTemplate.

---

**Message from the maintainers**:

Love this feature request? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by panicboat**
Fixes #13282

### Motivation

As indicated in the Issue, it is inconvenient that the history is not visible in the WorkflowTemplate details.
CronWorkflow can see it, so I would like to see consistency.

#### WorkflowTemplate

image
image
image
image

<details>
<summary>No history</summary>

image
image
image
image
</details>

#### ClusterWorkflowTemplate

image
image
image
image

<details>
<summary>No history</summary>

image
image
image
image
</details>

### Verification

- [x] no data
- [x] Only the relevant Workflow is displayed in the execution history, even if there are multiple WorkflowTemplates.
- [x] Displayed with no problem even with multiple data.

### CODE REVIEW

File: ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx [hunk 1]
```diff
@@ -83,70 +139,67 @@ export function ClusterWorkflowTemplateDetails({history, location, match}: Route
...
-                                            .delete(name)
-                                            .then(() => navigation.goto(uiUrl('cluster-workflow-templates')))
-                                            .then(() => setError(null))
-                                            .catch(setError);
-                                    }
-                                });
-                            }
-                        }
-                    ]
+                    items: getItems()
```
Comment (agilgur5) [inline ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx:142]: this doesn't seem necessary?

File: ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx [hunk 2]
```diff
@@ -83,70 +139,67 @@ export function ClusterWorkflowTemplateDetails({history, location, match}: Route
...
                 }
             }}>
             <>
-                <ErrorNotice error={error} />
-                {!template ? (
-                    <Loading />
-                ) : (
-                    <ClusterWorkflowTemplateEditor template={template} onChange={setTemplate} onError={setError} onTabSelected={setTab} selectedTabKey={tab} />
+                <>
```
Comment (agilgur5) [inline ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx:146]: I don't think this extra fragment is necessary either, there's already a fragment surrounding it.

**Review by agilgur5 — CHANGES_REQUESTED** [review]
This should look similar to #11811, it seems like there's some unnecessary changes here?

Also given how duplicative this is, we should probably move this out into a separate component rather than repeating it in 3 or 4 separate places. I did actually mention this in https://github.com/argoproj/argo-workflows/pull/11790#discussion_r1320704088, which was before it had too many duplications. We should do that now

**Comment by agilgur5** [comment]
Looks like #13452 was opened to supersede this? @panicboat please use the same PR and resolve comments on the PR instead of opening a new one.

Also if you do open a new one, please make sure to reference the old one. Creating a new one with no references is confusing at best, and entirely misses history and comments at worst.

**Comment by agilgur5** [comment]
reopened

**Comment by panicboat** [comment]
@agilgur5 I should have consulted you before creating a new PR.
It seems I am not authorized to ReOpen.
I can continue this PR if you ReOpen it.
image

**Comment by agilgur5** [comment]
> As to the main issue, I do not have the authority to reopen this PR.

Ah, yea in the future a maintainer can definitely reopen.

> Can you please reopen this PR so I can continue to work on it?

Ironically I now can't reopen this PR because the new one uses the same branch 🙃 
Screenshot 2024-08-13 at 6 06 10 PM


Let's just move to the new one then, it looks like you resolved the comments here in that one

File: ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx (hunk 2 above)
Comment (agilgur5) [reply ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx:146]: Resolved in #13452

File: ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx (hunk 1 above)
Comment (agilgur5) [reply ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx:142]: Resolved in #13452

**Pull request closed without merging**